src/Handler/CurlMultiHandler.php

```diff
@@ -49,25 +68,34 @@ public function __construct(array $options = [])
             $this->selectTimeout = 1;
         }
 
-        $this->options = isset($options['options']) ? $options['options'] : [];
+        $this->options = $options['options'] ?? [];
     }
 
+    /**
+     * @param string $name
+     *
+     * @return resource
+     */
     public function __get($name)
     {
-        if ($name === '_mh') {
-            $this->_mh = \curl_multi_init();
+        if ($name !== '_mh') {
+            throw new \BadMethodCallException("Can not get other property as '_mh'.");
+        }
 
-            foreach ($this->options as $option => $value) {
-                // A warning is raised in case of a wrong option.
-                curl_multi_setopt($this->_mh, $option, $value);
-            }
+        $multiHandle = \curl_multi_init();
+
+        if (false === $multiHandle) {
+            throw new \RuntimeException('Can not initialize curl multi handle.');
+        }
+
+        $this->_mh = $multiHandle;
 
-            // Further calls to _mh will return the value directly, without entering the
-            // __get() method at all.
-            return $this->_mh;
+        foreach ($this->options as $option => $value) {
+            // A warning is raised in case of a wrong option.
+            curl_multi_setopt($this->_mh, $option, $value);
```

Comment:
Does this intentionally not have a leading slash for mocking?

Reply:
`curl_multi_setopt` gets overwritten in `tests/bootstrap.php`. Adding the leading slash makes `\GuzzleHttp\Tests\Handler\CurlMultiHandlerTest::testCanAddCustomCurlOptions` fail.

Maybe make a note somewhere to change this test mock?